compiler/ir/backend.jvm/lower/src/org/jetbrains/kotlin/backend/jvm/lower/TypeOperatorLowering.kt

```diff
@@ -757,6 +757,9 @@ private class TypeOperatorLowering(private val backendContext: JvmBackendContext
         origin == IrDeclarationOrigin.DELEGATED_PROPERTY_ACCESSOR ||
                 origin == IrDeclarationOrigin.DELEGATED_MEMBER
 
+    private fun CharSequence.validSourcePosition(startOffset: Int, endOffset: Int): Boolean =
+        startOffset in 0 until endOffset && endOffset < length
```

Comment:
I don't understand... The change is that now we skip the case when `startOffset >= endOffset`? Isn't it an incorrect state for non-undefined offsets to be in?

Reply:
It is! And that is the state that's returned by the `extents` function in the case that it only meets offsets < 0 in the children of the element. In that case it returns `MAX_INT to 0` and it fails this check.

We could perhaps replace that part of the logic with an Option or null or something, if you'd prefer to not complicate the arithmetics?

Reply:
Oh, looking at it again I'm not sure what confused me the first time. It's clear this way too. Thanks for the clarification!